classifier/preprocess_cola_dataset_for_classifier.py

```diff
@@ -38,14 +38,15 @@ def main(args):
     sentences_positive = []
     sentences_negative = []
     
-    with open(data_file) as tsv_file:
-        read_tsv = csv.reader(tsv_file, delimiter="\t")
-        for line in read_tsv:
-            if int(line[1]) == 1:
-                sentences_positive.append(line[3])
-            else:
-                sentences_negative.append(line[3])
-
+    tsv_file = open(data_file)
+    read_tsv = csv.reader(tsv_file, delimiter="\t")
+    for line in read_tsv:
+        if int(line[1]) == 1:
+            sentences_positive.append(line[3])
+        else:
+            sentences_negative.append(line[3])
+    tsv_file.close()
+    
```

Comment:
Sorry about the confusion in my previous comment. 
The original with comment should be kept,
I just meant you can make the for loop be aligned to the with:

with open(data_file) as tsv_file:
        read_tsv = csv.reader(tsv_file, delimiter="\t")
for line in read_tsv:
        if int(line[1]) == 1:
            sentences_positive.append(line[3])
        else:
            sentences_negative.append(line[3])

Reply:
I don't think I can use read_tsv (like for line in read_tsv) outside the with open as it will throw  "I/O operation on closed file." error. So if I keep the with open structure, I guess I will have to keep the for loop inside the with open. Which structure would you recommend? Thanks!

Reply:
you can just keep your original version. Sorry I didn't realize read_tsv is used later.

Reply:
Ok! Thanks!